beacon-chain/blockchain/service.go

```diff
@@ -396,3 +542,82 @@ func (c *ChainService) calculateNewBlockVotes(block *types.Block, beaconState *t
 
 	return nil
 }
+
+func (c *ChainService) isBlockReadyForProcessing(block *types.Block) bool {
```

Comment:
These validity conditions should be moved to `core/blocks/`

Reply:
+1

Reply:
Are you planning to move this?

Reply:
The validity conditions have been moved to core, however we still have to fetch the parent block and POW chain reference block, so I am fetching the blocks over here. I didnt want to call service layer methods(`db.GetBlock()`, `PowClient.BlockByHash()`) from the core package. I can move those methods also to the core package if you guys want